src/contrast_gui.jl

```diff
@@ -1,6 +1,46 @@
 using Gtk.GConstants.GtkAlign: GTK_ALIGN_START, GTK_ALIGN_END, GTK_ALIGN_FILL
 
-function contrast_gui(enabled::Signal{Bool}, hist::Signal, clim::Signal)
+function change_red(col::CLim{T}, chan::CLim{T2}) where {T<:AbstractRGB, T2<:GrayLike}
+    cmin = T(chan.min, green(col.min), blue(col.min))
+    cmax = T(chan.max, green(col.max), blue(col.max))
+    return CLim(cmin, cmax)
+end
+function change_green(col::CLim{T}, chan::CLim{T2}) where {T<:AbstractRGB, T2<:GrayLike}
+    cmin = T(red(col.min), chan.min, blue(col.min))
+    cmax = T(red(col.max), chan.max, blue(col.max))
+    return CLim(cmin, cmax)
+end
+function change_blue(col::CLim{T}, chan::CLim{T2}) where {T<:AbstractRGB, T2<:GrayLike}
+    cmin = T(red(col.min), green(col.min), chan.min)
+    cmax = T(red(col.max), green(col.max), chan.max)
+    return CLim(cmin, cmax)
+end
+change_red(col::Signal, chan::CLim) = change_red(value(col), chan)
+change_green(col::Signal, chan::CLim) = change_green(value(col), chan)
+change_blue(col::Signal, chan::CLim) = change_blue(value(col), chan)
+
+function contrast_gui(enabled::Signal{Bool}, hists::Vector, clim::Signal{CLim{T}}) where {T<:AbstractRGB}
+    @assert length(hists) == 3 #one signal per color channel
+    chanlims = channel_clims(value(clim))
+    rsig = Signal(chanlims[1])
+    gsig = Signal(chanlims[2])
+    bsig = Signal(chanlims[3])
+    #make sure that changes to individual channels update the color clim signal and vice versa
+    bindmap!(clim, x->change_red(clim, x), rsig, x->channel_clim(red, x); initial = false)
```

Comment:
On the Travis run, `bindmap!` caused an error. Looks like `bindmap!` was merged to master in Reactive [in Sept 2017](https://github.com/JuliaGizmos/Reactive.jl/pull/161) but Reactive hasn't been tagged since August 2017. If you need this then you might want to nudge them?

Reply:
I nudged [here](https://github.com/JuliaGizmos/Reactive.jl/issues/176).  I'll wait a bit and see whether I get a response.  If not I can rework it to avoid using `bindmap!`.